File: programacao-em-python/Exercicios/banco/ContaBancaria.py
class ContaBancaria:
    def __init__(self):
        self._numero_conta = 0
        self._titular = ''
        self._saldo = 0.0
        self._ativo = False
    
    # Setters
    def set_numero_conta(self, numero_conta):
        self._numero_conta = numero_conta
    
    def set_titular(self, titular):
        self._titular = titular
    
    def set_saldo(self, saldo):
        self._saldo = saldo

    # Getters

    def get_numero_conta(self):
        return self._numero_conta
    
    def get_titular(self):
        return self._titular
    
    def get_ativo(self):
        return self._ativo


    # Métodos

    def ativar_conta(self):
        self._ativo = True
    
    def depositar(self, valor):
        self._saldo += valor

    def sacar(self, valor):
        if self._saldo >= valor:
            self._saldo -= valor
            return True
        return False
    
    def verificar_saldo(self):
        print(f'Saldo: R$ {self._saldo:.2f}')
        return

File: ProjetosPOO/ContasBancarias/src/principal/ContaBancaria.java
package principal;

/**
 *
 * @author wescleyj
 */

public class ContaBancaria {
    private String numeroConta;
    private String titular;
    private double saldo;
    
    // Contrutores
    // Construtor (sem parametros)
    public ContaBancaria() {
        numeroConta = "";
        titular = "";
        saldo = 0;
    }
    
    // Construtor (com parametros)
    public ContaBancaria(String numeroConta, String titular, double saldo) {
        this.numeroConta = numeroConta;
        this.titular = titular;
        this.saldo = saldo;
    }
    
    // Metodos
    public void depositar(double valor) {
        if(valor > 0) {
            saldo += valor;
            System.out.println("Valor adicionado com sucesso");
        }
        else {
            System.out.println("Saldo invalido");
        }
    }
    
    public void sacar(double valor) {
        if(valor <= saldo) {
            saldo -= valor;
            System.out.println("Você sacou com sucesso: R$" + valor);
        }
        else {
            System.out.println("O saldo atual em conta é inferior ao valor solicitado, transação não concluida");
        }
    }
    
    @Override
    public String toString() {
        return "Conta Bancaria {" +
               "Numero da conta = '" + numeroConta + '\'' +
               " Titular da conta = '" + titular + '\'' +
               " Saldo da conta = '" + saldo + '\'' +
               '}';
    }

    public String getNumeroConta() {
        return numeroConta;
    }

    public String getTitular() {
        return titular;
    }

    public double getSaldo() {
        return saldo;
    }

    public void setSaldo(double saldo) {
        this.saldo = saldo;
    }
}
